src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -1650,11 +1650,33 @@ void ScriptMgr::OnPlayerRemoveFromBattleground(Player* player, Battleground* bg)
     FOREACH_SCRIPT(PlayerScript)->OnRemoveFromBattleground(player, bg);
 }
 
+bool ScriptMgr::OnBeforeAchievementComplete(Player* player, AchievementEntry const* achievement)
```

Comment:
wouldnt this be OnAchievementInProgress?

Reply:
https://www.azerothcore.org/wiki/hooks-script#naming-conventions

The naming convention we should follow from now on implies this structure: On<When><Action>

normally the Action refers to the parent function (in this case AchievementComplete), while the "When" refers in which part of the function the hooks is called. If we follow the ["Single responsibility principle"](https://en.wikipedia.org/wiki/Single-responsibility_principle), it's very unlikely to have more than 2 hooks per each function (OnBefore<action> and OnAfter<action>). However, edge cases always exist and it's fine to break the convention sometimes if we document our hooks properly. However, in this case, it's not needed.